modules/hello-world/examples/DocParser.java

```diff
@@ -0,0 +1,338 @@
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Files;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Stack;
+
+public class DocParser {
+
+
+	int fileNo = 0;
+	BufferedWriter codeWriter;
+
+	public static void main(String[] args) throws IOException {
+		DocParser parser = new DocParser();
+		for (String fileName : args) {
+			try {
+				parser.process(fileName);
+			} catch (IOException e) {
+				System.err.println(e);
+			}
+		}
+	}
+
+	String preamble=
```

Comment:
stringbuilder maybe?  This isn't so idiomatic java, is it?

Reply:
I could be wrong, so don't take that as a strong statement.  :)

Reply:
String concatenation is fine here. StringBuilder only adds value when the string is assembled using branches or loops.